src/Sort.ts

```diff
@@ -1,8 +1,46 @@
 import type { Task } from './Task';
+import type { Query } from './Query';
+import type moment from 'moment';
+
+type Comparator = (a: Task, b: Task) => -1 | 0 | 1;
 
 export class Sort {
-    public static byStatusThenDateThenPath(tasks: Task[]): Task[] {
-        return tasks.sort(Sort.compareByStatus);
+    public static by(query: Pick<Query, 'sorting'>, tasks: Task[]): Task[] {
+        const priorities: Comparator[] = [
+            this.compareByStatus,
+            this.compareByDueDate,
+            this.compareByPath,
+        ];
+
+        for (const sortProp of query.sorting.reverse()) {
+            switch (sortProp) {
+                case 'status':
+                    priorities.unshift(this.compareByStatus);
+                    break;
+                case 'due':
+                    priorities.unshift(this.compareByDueDate);
+                    break;
+                case 'done':
+                    priorities.unshift(this.compareByDoneDate);
+                    break;
+            }
+        }
```

Comment:
This means you do a little more sorting than necessary, but I think that should be fine. You would sort by `status` twice, for example. But as I said, I think it's ok this way.

Reply:
yeah, also thought about that. but since sorting works through the properties [one-by-one](https://github.com/Skn0tt/obsidian-tasks/blob/d869b95046dd6ce32dd2cc9bd1ec490ebe01c866/src/Sort.ts#L39-L39), double-sorting will happen only for *very* similar items. and since this is only about speed, not about sorting correctness (sorting twice by the same property won't change anything, no matter the sort order), I think it should be fine :D